src/query/service/src/pipelines/processors/transforms/aggregator/transform_partition_bucket.rs

```diff
@@ -215,10 +230,72 @@ impl<Method: HashMethodBounds, V: Copy + Send + Sync + 'static>
 
         if self.max_partition_count > 0 {
             let meta = data_block.take_meta().unwrap();
-            if let Some(AggregateMeta::AggregateHashTable(p)) =
-                AggregateMeta::<Method, V>::downcast_from(meta)
-            {
-                self.partition_payloads.push(p);
+            if let Some(block_meta) = AggregateMeta::<Method, V>::downcast_from(meta) {
+                return match block_meta {
+                    AggregateMeta::AggregatePayload(p) => {
+                        let res = p.bucket;
+                        self.agg_payloads.push(p);
+                        res
+                    }
+                    AggregateMeta::Serialized(p) => {
+                        let rows_num = p.data_block.num_rows();
+                        let radix_bits = p.max_partition_count.trailing_zeros() as u64;
+                        let config = HashTableConfig::default().with_initial_radix_bits(radix_bits);
+                        let mut state = ProbeState::default();
+                        let capacity = AggregateHashTable::get_capacity_for_count(rows_num);
+                        let mut hashtable = AggregateHashTable::new_with_capacity(
+                            self.params.group_data_types.clone(),
+                            self.params.aggregate_functions.clone(),
+                            config,
+                            capacity,
+                        );
+
+                        let agg_len = self.params.aggregate_functions.len();
+                        let group_len = self.params.group_columns.len();
+                        let agg_states = (0..agg_len)
+                            .map(|i| {
+                                p.data_block
+                                    .get_by_offset(i)
+                                    .value
+                                    .as_column()
+                                    .unwrap()
+                                    .clone()
+                            })
+                            .collect::<Vec<_>>();
+                        let group_columns = (agg_len..(agg_len + group_len))
+                            .map(|i| {
+                                p.data_block
+                                    .get_by_offset(i)
+                                    .value
+                                    .as_column()
+                                    .unwrap()
+                                    .clone()
+                            })
+                            .collect::<Vec<_>>();
+
+                        let _ = hashtable
+                            .add_groups(
```

Comment:
We don't need probe the HT.  Just save the groups into payload.

So you should add a new method to convert groups into payload in HT.